File: test/napi/lib/node.js
var addon = require("..");
var assert = require("chai").assert;

describe("Node Standard Library", function () {
  it("should print to console", function () {
    addon.call_console_log_and_error();
  });

  it("process.versions.node", function () {
    assert.strictEqual(addon.get_node_version(), process.versions.node);
  });
});

File: test/napi/src/js/node.rs
use neon::node::Node;
use neon::prelude::*;

pub fn call_console_log_and_error(mut cx: FunctionContext) -> JsResult<JsUndefined> {
    cx.console().log("This is cx.console().log()!")?;
    cx.console().error("This is cx.console().error()!")?;
    Ok(cx.undefined())
}

pub fn get_node_version(mut cx: FunctionContext) -> JsResult<JsString> {
    let versions = cx.process().versions()?;
    let data = versions.data();
    Ok(cx.string(&data.node))
}
